internal/batches/watchdog/watchdog.go

```diff
@@ -0,0 +1,30 @@
+package watchdog
+
+import (
+	"time"
+
+	"github.com/derision-test/glock"
+)
+
+type WatchDog struct {
+	ticker   glock.Ticker
+	callback func()
+}
+
+func New(interval time.Duration, callback func()) *WatchDog {
+	ticker := glock.NewRealTicker(interval)
+	return &WatchDog{
+		ticker:   ticker,
+		callback: callback,
+	}
+}
+
+func (w *WatchDog) Stop() {
+	w.ticker.Stop()
+}
+
+func (w *WatchDog) Start() {
+	for _ = range w.ticker.Chan() {
```

Comment:
I realised while re-reviewing this that we can't do it quite this simply — I vaguely remembered that there was some weirdness around something to do with channels, and it was that [stopped Tickers are not actually closed](https://pkg.go.dev/time#Ticker.Stop), because… well, you can read the docs. (I honestly interpret it mostly as "because fuck you, that's why".)

This would have been OK if it was a "normal" channel that was closed when it was no longer needed. Bah.

So, two options:

1. We reinstate the previous, two channel behaviour, just with a `return` on the `done` case, or
2. We decide we don't care about leaking a goroutine, because `src` is likely about to exit anyway.

I'm not super fussed either way. I'm sorry for forgetting the nuance until after you'd done the work!

Reply:
There's no need to apologize. This ticket has been a big learning experience with channels for me - learnt a lot from pairing with @Piszmog earlier.

I actually read the docs and saw the comment about the channel not being closed after calling Stop() but I figured it wasn't going to be an issue since src will edit anyway.

However now that I think about it, if src ever tried doing some background work like we do for sg (to check for an update whenever you run a command), then this could lead to an issue.

So I'll reinstate the former way I did it tomorrow, to avoid this. Thanks for taking a closer look @LawnGnome